oaknational/oak-ai-lesson-assistant
# feat: invalidate moderation and unblock user

## Description

- ~adds `/api/admin/...` route to invalidate a moderation~
- adds trpc admin procedure and router
- adds admin page for chat with moderations on, with button to invalidate a moderation 
- adds methods to safety violations class to revert a safety violation
- adds method to moderations class to invalidate a moderation
- updates moderation slack notification with a link to the 'invalidate moderation' link
- adds relevant migration (which has been run against staging)

**ps: excuse the styles, it's internal right?**

## Issue(s)

https://www.notion.so/oaknationalacademy/Be-able-to-de-block-a-lesson-that-was-deemed-toxic-cf42d50e12b64d649bd95207aaac5baf?pvs=4

## How to test

1. Go to the app
2. make a lesson and get a toxic moderation e.g. using mod:tox
3. go to #ai-dev-user-events in slack
4. click "Admin chat"
5. click invalidate next to your moderation
6. you should have the moderation invalidated and the chat be available again

**Note: you can also get yourself banned and test that it unbans you **

## Screenshots

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/e48c1b35-b965-441f-9260-f1153ad1238c">
